### ISSUE TITLE
Learning rate issue

### ISSUE THREAD

**Issue by olix86** [ivadomed/ivadomed#57]
It looks like the LR currently only does a single annealing cycle : 

![image](https://user-images.githubusercontent.com/5721601/66519682-c5cc6e00-eab5-11e9-98f5-8a6aec724df8.png)

I would have expected the following for cosine annealing : 
![image](https://user-images.githubusercontent.com/5721601/66519748-e1377900-eab5-11e9-8dad-6944c45d3752.png)

I'm unsure if we're calling the function correctly since it looks like `T_max` ([pytorch docs](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.CosineAnnealingLR)) should be the period of a single cycle and not the entire training process as we're doing : 

`scheduler = optim.lr_scheduler.CosineAnnealingLR(optimizer, num_epochs)`

The scheduler was modified in #55 and #51 but these are not the cause of the issue, as I've reverted to before both PR (7c36f943128225d3886862ca66fee889e352ee7e) and the situation is the same.

Surprisingly, it looks like the dice was better when a bug was introduced, between these 2 PR, which updated the LR at each batch instead of each epoch, thereby unintentionally doing multiple cycles during training : 

![image](https://user-images.githubusercontent.com/5721601/66584501-45f4e100-eb53-11e9-8445-852de0cda228.png)

  
A simple workaround to reproduce this would be
`scheduler = optim.lr_scheduler.CosineAnnealingLR(optimizer, num_epochs/num_cycles)`

Which has an example would give for `num_cycles = 10`

![image](https://user-images.githubusercontent.com/5721601/66582323-95d1a900-eb4f-11e9-9a55-5e0995df458c.png)

However a true cosine annealing ([paper](https://arxiv.org/abs/1608.03983)) requires restarts, which are not implemented in pytorch, but there are 3rd party implementations : [link](https://github.com/mpyrozhok/adamwr)

I'm unsure what's the best way to solve this going forward, as I don't know what was the initial intention when this was implemented. 

@charleygros what do you think? :)
